lib/tasks/asset_manager.rake

```diff
@@ -0,0 +1,33 @@
+namespace :asset_manager do
+  desc "Report on attachments that are only attached to non-visible editions in Whitehall but not deleted in Asset Manager"
+  task report_deleted_assets: :environment do
+    output = AttachmentData.find_each.map do |attachment_data|
+      attachables = attachment_data.attachments.map(&:attachable).compact
+
+      next unless attachables.any?
+
+      non_edition_attachables = attachables.reject do |attachable|
+        attachable.is_a?(Edition)
+      end
+
+      next if non_edition_attachables.any?
```

Comment:
Given `non_edition_attachables` isn't used after this, you could simplify this and allow early exits from the iterator:

```rb
next if attachables.detect { |attachable| !attachable.is_a?(Edition) }
```

Reply:
Nice.  I was using this variable later on, but didn't need it in the end as I realised these were out of scope for the problem.